src/repart/repart.c

```diff
@@ -2690,9 +2701,8 @@ static int context_copy_from_one(Context *context, const char *src) {
         if (r < 0)
                 return r;
 
-        r = fd_verify_regular(fd);
-        if (r < 0)
-                return log_error_errno(r, "%s is not a file: %m", src);
+        if (!verify_regular_or_block(fd))
```

Comment:
pleas epropagate the original error, if we have one. i.e. if < 0 return return value as is.

Reply:
Sure, would be worth propagating the error if returned from `fstat`, but what should I return if `!(ISREG || ISBLK)`?

Reply:
the same errors that stat_verify_regular() returns...